rust/flatbuffers/src/builder.rs

```diff
@@ -754,3 +832,147 @@ impl<'fbb> Default for FlatBufferBuilder<'fbb> {
         Self::with_capacity(0)
     }
 }
+
+/// An index that indexes from the reverse of a slice.
+///
+/// Note that while the internal representation is an inde
+/// from the end of a buffer, operations like `Add` and `Sub`
+/// behave like a regular index:
+///
+/// # Examples
+///
+/// ```ignore
+/// let buf = [0, 1, 2, 3, 4, 5];
+/// let idx = ReverseIndex::back() - 2;
+/// assert_eq!(&buf[idx.range_to_end()], &[4, 5]);
+/// assert_eq!(idx.to_forward_index(&buf), 4);
+/// ```
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+struct ReverseIndex(usize);
+
+impl ReverseIndex {
+    /// Returns an index set to the end.
+    ///
+    /// Note: Indexing this will result in an out of bounds error.
+    pub fn back() -> Self {
```

Comment:
optional nit: Maybe I've been writing too much cpp, but I prefer `end` instead of `back` since in that language, they mean "1 past the last element" and "the last element" respectively.

Reply:
Yeah, that's reasonable. `back` implies the last element. Done.